src/modules/facility-loan-transaction/facility-loan-transaction.service.ts

```diff
@@ -0,0 +1,50 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { AcbsAuthenticationService } from '@ukef/modules/acbs-authentication/acbs-authentication.service';
+import { DateStringTransformations } from '@ukef/modules/date/date-string.transformations';
+
+import { AcbsFacilityLoanTransactionService } from '../acbs/acbs-facility-loan-transaction.service';
+import { GetFacilityLoanTransactionResponseItem } from './dto/get-loan-transaction-response.dto';
+
+@Injectable()
+export class FacilityLoanTransactionService {
+  constructor(
+    private readonly acbsAuthenticationService: AcbsAuthenticationService,
+    private readonly acbsFacilityLoanTransactionService: AcbsFacilityLoanTransactionService,
+    private readonly dateStringTransformations: DateStringTransformations,
+  ) {}
+
+  async getLoanTransactionsByBundleIdentifier(bundleIdentifier: string): Promise<GetFacilityLoanTransactionResponseItem> {
+    const idToken = await this.acbsAuthenticationService.getIdToken();
+    const loanTransaction = await this.acbsFacilityLoanTransactionService.getLoanTransactionByBundleIdentifier(bundleIdentifier, idToken);
+    const loan = loanTransaction.BundleMessageList[0];
+    if (loan.$type !== 'NewLoanRequest') {
```

Comment:
Can we have a constant for `NewLoanRequest` please?

Reply:
If we do this, we seem to lose the fact that the type system can then recognise that we have a loan with a type of `NewLoanRequest` for some reason.